jobs/jobs.go

```diff
@@ -43,14 +86,45 @@ func (mgr *JobManager) AddJobs(jobs ...Job) {
 func (mgr *JobManager) AddJob(j Job) {
 	mgr.wgJobs.Add(1)
 	go func() {
-		defer mgr.wgJobs.Done()
-		j(mgr.errChan)
+		if err := mgr.jobCapacity.Acquire(mgr.ctx, 1); err != nil {
+			mgr.tryCacheError(err)
+			mgr.jobCapacity.Release(1)
```

Comment:
If we failed to acquire, do we need to release?

Reply:
that was supposed to be waitgroup.Done(), good catch!